test/utils.ts

```diff
@@ -97,15 +99,41 @@ export function simplify(latex: string): string {
   return exprToString(engine.parse(latex)?.simplify());
 }
 
-export function checkJson(inExpr: SemiBoxedExpression | null): string {
+type ExprVariant =
```

Comment:
I ended up not using this (as implemented in 'checkJson') in the end, due to using a custom solution in *canonical-form.test.ts* which prints a 'canonical-forms' variant: but conceivably, could see myself as having a use for this.
If not your cup of tea, do do away!

Reply:
Looks good.